diff --git a/core/ghost.js b/core/ghost.js
index 86735d2..fdc36c3 100644
--- a/core/ghost.js
+++ b/core/ghost.js
@@ -337,12 +337,12 @@ Ghost.prototype.doFilter = function (name, args, callback) {
         });
     });

-    callback(args);
+    return callback(args);
 };

 // Initialise plugins.  Will load from config.activePlugins by default
 Ghost.prototype.initPlugins = function (pluginsToLoad) {
-    pluginsToLoad = pluginsToLoad || models.Settings.activePlugins;
+    pluginsToLoad = pluginsToLoad || config.activePlugins || [];
     var self = this;

     return plugins.init(this, pluginsToLoad).then(function (loadedPlugins) {
diff --git a/core/server.js b/core/server.js
index 16ffb0c..85ec9f3 100644
--- a/core/server.js
+++ b/core/server.js
@@ -21,6 +21,7 @@ var express     = require('express'),
     loading = when.defer(),
     server = express(),
     ghost = new Ghost();
+    ghost.server = server;

 // ##Custom Middleware

